### ISSUE TITLE
Exception not raised by the KafkaConsumer() method for invalid address

### ISSUE THREAD

**Issue by aliartiza75** [dpkp/kafka-python#1481]
Hello everyone,

In the earlier version of the Kafka-python module, the method KafkaConsumer() when received an invalid broker address used to raise an exception but in the latest version, it neither raises an exception nor timeouts but stucks at that point. 
Module version is : 1.4.2

```
client = KafkaConsumer(bootstrap_servers=broker_addr, request_timeout_ms=6000, session_timeout_ms=5000)
```
Need help regarding the behavior of this method.

Thanks in advance

**Comment by affansyed** [comment]
+1. This is breaking lots of unittests at our place.

**Comment by jeffwidman** [comment]
When you say invalid, what kind of invalid? 

Is it a validly formed host that simply doesn't have a running broker or a completely malformed hostname that will never work?

Related: https://github.com/dpkp/kafka-python/issues/1473

**Comment by aliartiza75** [comment]
Malformed addresses example are given below:

- "invalid_addresss"
- ""444.444.22.11" 

and yes it relates to #1473

**Comment by dpkp** [comment]
Fixed via #1507
